apps/documentation/src/components/TableOfContents/TableOfContents.tsx

```diff
@@ -1,115 +1,115 @@
+        rootMargin: '-50px 0px -80% 0px',
+        threshold: [0.1, 0.5, 1.0],
+      },
+    )
 
-export function TableOfContents({
-  withTabs,
-}: TableOfContentsProps): JSX.Element | null {
-  const [active, setActive] = useState(0)
-  const [headings, setHeadings] = useState<Array<Heading>>([])
-  const headingsRef = useRef<Array<Heading>>([])
-  const router = useRouter()
+    headingsRef.current.forEach((node) => {
+      observer.observe(node)
+    })
+    observerRef.current = observer
 
-  const filteredHeadings = headings.filter((heading) => heading.depth > 1)
+    const handleHashChange = (): void => {
+      setTimeout(() => {
+        observerRef.current?.disconnect()
+        observerRef.current = observer
+        headingsRef.current.forEach((node) => {
+          observer.observe(node)
+        })
+      }, 300)
+    }
 
-  const handleScroll = (): void => {
-    setActive(
-      getActiveElement(
-        headingsRef.current.map((d) => d.getNode().getBoundingClientRect()),
-      ),
-    )
-  }
+    window.addEventListener('hashchange', handleHashChange)
 
-  useEffect(() => {
-    const _headings = getHeadings()
-    headingsRef.current = _headings
-    setHeadings(_headings)
-    setActive(
-      getActiveElement(
-        _headings.map((d) => d.getNode().getBoundingClientRect()),
-      ),
-    )
-    window.addEventListener('scroll', handleScroll)
     return (): void => {
-      window.removeEventListener('scroll', handleScroll)
+      observer.disconnect()
+      window.removeEventListener('hashchange', handleHashChange)
     }
   }, [router.pathname])
 
-  if (filteredHeadings.length === 0) {
-    return null
+  const handleClick = (
+    event: MouseEvent<HTMLAnchorElement>,
+    id: string,
+  ): void => {
+    event.preventDefault()
+    const element = document.getElementById(id)
+    if (element) {
+      element.scrollIntoView({
+        behavior: 'instant',
+        block: 'start',
+      })
+    }
   }
 
-  const items = filteredHeadings.map((heading, index) => (
-    <Text
-      key={heading.id}
-      component={Link}
-      fz="sm"
-      p={10}
-      className={classes.link}
-      mod={{ active: active === index }}
-      href={`#${heading.id}`}
-      __vars={{ '--toc-link-offset': `${heading.depth - 1}` }}
-    >
-      {heading.content}
-    </Text>
-  ))
+  // Avoid to show it in case of a TODO page
+  if (headings.length === 1) {
+    return null
+  }
 
   return (
-    <Box
-      component="nav"
-      mod={{ 'with-tabs': withTabs }}
-      className={classes.wrapper}
-    >
+    <Box component="nav" className={classes.wrapper}>
       <div className={classes.inner}>
         <div>
-          <div className={classes.header}>
-            <IconList
-              style={{ width: rem(20), height: rem(20) }}
-              stroke={1.5}
-            />
-            <Text className={classes.title}>Table of contents</Text>
+          <Text className={classes.title} mb={8}>
+            On this page
```

Comment:
<img width="212" alt="image" src="https://github.com/user-attachments/assets/44323157-5807-472d-9ba3-ca61765bfdfc" />

When scrolling the page "On this page" title has no spacing from it and the main header